lib/message/utils.js

```diff
@@ -63,26 +75,30 @@ export async function getCleartextMessage(message) {
 }
 
 /**
- * Create a cleartext message from a text.
- * @param {String|openpgp.cleartext.CleartextMessage} message
- * @return {openpgp.cleartext.CleartextMessage}
+ * Get a signed message from the given data.
+ * Either `textData` or `binaryData` must be specified.
+ * @param {Object} options - input for openpgp.sign
+ * @param {String|ReadableStream<String>} textData - text data to sign
+ * @param {Uint8Array|ReadableStream<Uint8Array>} binaryData - binary data to sign
+ * @param {Boolean} stripTrailingSpaces - whether trailing spaces should be removed from `textData`
+ * @returns Promise<{Message|Signature|String|ReadableStream<String>}> signed message object, signature object, or their corresponding armored data
+ * @throws on signing error
  */
-export function createCleartextMessage(message) {
-    if (CleartextMessage.prototype.isPrototypeOf(message)) {
-        return message;
+export async function signMessage({
+    textData,
+    binaryData,
+    stripTrailingSpaces,
+    date = serverTime(),
+    armor = true,
+    ...options
+}) {
+    const dataType = binaryData ? 'binary' : 'text';
+    const data = binaryData || (stripTrailingSpaces ? removeTrailingSpaces(textData) : textData); // throw if streamed text and stripTrailingSpaces enabled
+    const sanitizedOptions = {
+        ...options,
+        date,
+        message: await createMessage({ [dataType]: data })
     }
-    return openpgpCreateCleartextMessage({ text: message });
-}
-
-export function createMessage(source, filename, date = serverTime()) {
-    if (Uint8Array.prototype.isPrototypeOf(source)) {
-        return openpgp_createMessage({ binary: source, filename, date });
-    }
-    return openpgp_createMessage({ text: source, filename, date });
-}
```

Comment:
https://github.com/ProtonMail/pmcrypto/pull/133#discussion_r793833253 also means that we should probably keep this wrapper function, actually, even if it's just for the `date = serverTime()` default :/